### ISSUE TITLE
Pass through USB TPM devices into Earthly image for GPG signing

### ISSUE THREAD

**Issue by hongkongkiwi** [earthly/earthly#3319]
**What existing functionality needs improvement?**

It would be great to be able to pass in USB TPM devices for GPG signing into Earthly.

When using hardware based TPM devices such as Yubikey for code signing and encryption. We can automate this with docker and there is an [ability to pass in USB devices into docker to make this work](https://github.com/bernadinm/docker-yubikey). 

Doing this in a cross platform way is a little bit tricky, there's an open bug request hyperkit for this: https://github.com/moby/hyperkit/issues/149 and I'm not clear on the status.

For organisations which requires code signing, this would allow them to have a reproducible way to do this via Earthly which I think is a big win. It would great to have earthly's help to resolve this.

This also applies to performing functions on crypto hardware devices via docker containers via earthly which I think is also a big use case.

**Expected Behavior**

Bring in USB devices into layers, specifically TPM devices for GPG signing in a way that works cross platform.

**Comment by alexcb** [comment]
We will have to investigate the feasibility of mounting host volumes to support this specific request for passing through a USB device.

earthly already supports passing an ssh socket from the host into a container by using `RUN --ssh`; and I see that `gpg-agent` creates a socket under `~/.gnupg/S.gpg-agent`, it might be possible to use the existing socket passing code to pass along this socket.

I see that yubikey can be integrated with GnuPG: https://support.yubico.com/hc/en-us/articles/360013790259-Using-Your-YubiKey-with-OpenPGP

Do you know if this could work with your requirements?

**Comment by alexcb** [comment]
> Do you have a working example with Earthly? I'm not clear from the above link how this would be integrated into Earthly.

No. I was only proposing a potential solution, which would likely be easier for earthly to implement.

But this solution depends on your ability to use the yubikey via the gpg agent socket. If you are unable to use the gpg agent for your signing requirement, then there's no point investigating if earthly could pass through the gpg agent.

**Comment by ejstreet** [comment]
> earthly already supports passing an ssh socket from the host into a container by using `RUN --ssh`; and I see that `gpg-agent` creates a socket under `~/.gnupg/S.gpg-agent`, it might be possible to use the existing socket passing code to pass along this socket.

My team would be interested in this being implemented
